src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/ListenerContext.cs

```diff
@@ -42,14 +42,22 @@ protected UvStreamHandle CreateAcceptSocket()
             }
         }
 
-        public PipeOptions LibuvPipeOptions => new PipeOptions
+        public PipeOptions LibuvInputPipeOptions => new PipeOptions
         {
             ReaderScheduler = ServiceContext.ThreadPool,
             WriterScheduler = Thread,
             MaximumSizeHigh = ServiceContext.ServerOptions.Limits.MaxRequestBufferSize ?? 0,
             MaximumSizeLow = ServiceContext.ServerOptions.Limits.MaxRequestBufferSize ?? 0
         };
 
+        public PipeOptions LibuvOutputPipeOptions => new PipeOptions
+        {
+            ReaderScheduler = Thread,
+            WriterScheduler = ServiceContext.ThreadPool,
+            MaximumSizeHigh = ServiceContext.ServerOptions.Limits.MaxResponseBufferSize ?? 0,
+            MaximumSizeLow = ServiceContext.ServerOptions.Limits.MaxResponseBufferSize ?? 0
+        };
+
         public PipeOptions AdaptedPipeOptions => new PipeOptions
```

Comment:
This needs some tweaking. We don't want request limits for both pipes in the adapter.